js/viz/chart.js

```diff
@@ -371,31 +371,38 @@ function collectMarkersInfoBySeries(allSeries, filteredSeries, argAxis) {
     return { series, overloadedSeries };
 }
 
-function applyAutoHidePointMarkers(allSeries, filteredSeries, overloadedSeries, argAxis) {
-    const argAxisType = argAxis.getOptions().type;
-    filteredSeries.forEach(s => {
-        const seriesIndex = allSeries.indexOf(s);
-        s.autoHidePointMarkers = false;
-        const tickCount = argAxis.getTicksValues().majorTicksValues.length;
-        if(s.autoHidePointMarkersEnabled() && (argAxisType === DISCRETE || overloadedSeries[seriesIndex].pointsCount > tickCount)) {
-            for(const index in overloadedSeries[seriesIndex]) {
-                const i = parseInt(index);
-                if(isNaN(i) || overloadedSeries[seriesIndex].total / overloadedSeries[seriesIndex].continuousSeries < 3) {
-                    continue;
-                }
-                if(i === seriesIndex) {
-                    if(overloadedSeries[i][i] * 2 >= overloadedSeries[i].pointsCount) {
-                        s.autoHidePointMarkers = true;
-                        break;
-                    }
-                } else if(overloadedSeries[seriesIndex].total >= overloadedSeries[seriesIndex].pointsCount) {
-                    s.autoHidePointMarkers = true;
-                    break;
-                }
+const checkOverlay = (overlayPoint, currentPoint, radiusPoint) => {
+    const leftBoundaryOverlay = overlayPoint.x - radiusPoint <= currentPoint.x;
+    const rightBoundaryOverlay = overlayPoint.x + radiusPoint >= currentPoint.x;
+    const topBoundaryOverlay = overlayPoint.y - radiusPoint <= currentPoint.y;
+    const bottomBoundaryOverlay = overlayPoint.y + radiusPoint >= currentPoint.y;
+
+    const isOverlayX = leftBoundaryOverlay && rightBoundaryOverlay;
+    const isOverlayY = topBoundaryOverlay && bottomBoundaryOverlay;
+
+    return isOverlayX && isOverlayY;
+};
+
+const cycleComparison = (startCounter, points, currentPoint, skippingSamePoints = true) => {
+    const radiusPoint = currentPoint._options.size / 2;
+
+    for(let i = startCounter; i < points.length; i++) {
+        if(!skippingSamePoints) {
+            const isSameCoordinateX = points[i].x === currentPoint.x;
+            const isSameCoordinateY = points[i].y === currentPoint.y;
+
+            if(isSameCoordinateX && isSameCoordinateY) {
+                continue;
```

Comment:
Why do we need `continue` here? To not check `checkOverlay` condition below? In this case i think we can just save `isSameCoordinateX && isSameCoordinateY` in a variable with meaningful name and use it in appropriate condition

Reply:
After a joint check with you, I changed this